experiment/src/org/labkey/experiment/samples/UploadSamplesHelper.java

```diff
@@ -441,25 +522,56 @@ else if (parts[0].equalsIgnoreCase(ExpData.DATA_OUTPUT_CHILD))
             }
             if (currentParents.second != null)
             {
+                boolean isExistingAliquot = false;
+                if (runItem instanceof ExpMaterial)
+                {
+                    ExpMaterial currentMaterial = (ExpMaterial) runItem;
+                    isExistingAliquot = !StringUtils.isEmpty(currentMaterial.getAliquotedFromLSID());
+
+                    if (isExistingAliquot && !isAliquot)
+                        throw new ValidationException("AliquotedFrom is absent for aliquot: " + currentMaterial.getName());
+                    else if (!isExistingAliquot && isAliquot)
+                        throw new ValidationException("Unable to change sample to aliquot: " + currentMaterial.getName());
+                    else if (isExistingAliquot)
+                    {
+                        if (!currentMaterial.getAliquotedFromLSID().equals(aliquotParent.getLSID()))
+                            throw new ValidationException("Aliquot parent cannot be updated");

```

Comment:
```suggestion
                            throw new ValidationException(Strring.format("Aliquot parents cannot be updated.  Change parent value from %s to %s.",currentMaterial.getAliquotedFromLSID(), aliquotParent.getLSID()));
```
This should help a user find the offending row in the input.